### PR TITLE
Styling structure guide

### PR DESCRIPTION
**Pull request by jgoley**
Establishes the basic structure of the styles package and adds a style guide.

I can't seem to import *Typographic* which is included in the stylus package ([mquandalle:stylus](https://atmospherejs.com/mquandalle/stylus)). Not sure why.

### CODE REVIEW

File: packages/styles/README.md [hunk 1]
```diff
@@ -1,3 +1,90 @@
...
+  - Importing: `@import viewname.import`
+  - Add to package.js: `api.addFiles('viewname.import.styl');`
+- Custom mixins should be added to *mixins.import.styl*
+- Global styles should be added to *globals.inport.styl*
+- Add global variables to *varaibles.import.styl*
+
+####Syntax Patterns
+- Do not use brackets, colons, or semicolons
+- Class names should be lowercase, hyphenated and as short as possible while accurately indicating purpose
+- Try to use classes as selectors only
```
Comment (aslagle) [inline packages/styles/README.md]: Do you mean use classes as selectors and don't use classes for other things, or use classes as selectors but don't use ids or tags as selectors?

File: packages/styles/README.md [hunk 2]
```diff
@@ -1,3 +1,90 @@
...
+- **Mixin Library**: [Nib](http://tj.github.io/nib/)
+
+###Adding Files and styles
+- Create a file with the name of associated view or package.  Import files into *main.styl* **and** add them to package.js:
+  - Filename: *viewname.import.styl*
+  - Importing: `@import viewname.import`
+  - Add to package.js: `api.addFiles('viewname.import.styl');`
+- Custom mixins should be added to *mixins.import.styl*
+- Global styles should be added to *globals.inport.styl*
+- Add global variables to *varaibles.import.styl*
```
Comment (aslagle) [inline packages/styles/README.md]: A couple typos here

File: packages/styles/extends.import.styl [hunk 3]
```diff
@@ -0,0 +1,15 @@
+.header-font
+  font-family 'Oswald', sans-serif
+
+.body-font
+  font-family 'Open Sans', sans-serif
+
+.icon
+  font-family FontAwesome
+  -webkit-font-smoothing antialiased
```
Comment (aslagle) [inline packages/styles/extends.import.styl]: Is this browser prefix an exception to the nib/autoprefixer pattern?

File: packages/styles/globals.import.styl [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+  margin-top 0
+  font-size 40px
+
+h2 a.more-info
+  font-size 12px
+  color $btn-default
+
+h3
+  margin-top 0
+  margin-bottom 15px
```
Comment (aslagle) [inline packages/styles/globals.import.styl]: why do we have margins on h2 and h3 but not h1 or h4

File: packages/styles/README.md [hunk 5]
```diff
@@ -1,3 +1,90 @@
...
+
+####Syntax Patterns
+- Do not use brackets, colons, or semicolons
+- Class names should be lowercase, hyphenated and as short as possible while accurately indicating purpose
+- Try to use classes as selectors only
+- Use single quotes
+- Prefix variable names and functions with `$`
+- Browser prefixes are added with Nib and Autoprefixer
```
Comment (aslagle) [inline packages/styles/README.md]: Put selectors on separate lines instead of comma-separating them?

File: packages/styles/README.md (hunk 1 above)
Comment (jgoley) [inline packages/styles/README.md]: The latter. I'll clarify.

File: packages/styles/extends.import.styl (hunk 3 above)
Comment (jgoley) [inline packages/styles/extends.import.styl]: I think font-smoothing is only available in webkit bowsers and firefox. I'll try changing it to `font-smoothing` and see if nib adds the prefixes.

File: packages/styles/globals.import.styl [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+    vAlign()
+    padding 0 0 0 5px
+    font-size 110%
+    line-height 0
+
+.btn-primary
+  makeButton($secondary)
+
+.btn-success
+.at-btn
```
Comment (aslagle) [inline packages/styles/globals.import.styl]: .at-btn is accounts-specific, right? probably shouldn't be in globals

File: packages/styles/globals.import.styl (hunk 4 above)
Comment (jgoley) [inline packages/styles/globals.import.styl]: I'm sure there was reasoning in GRRS. I'll review this so we start with a cleaner slate.

File: packages/styles/README.md (hunk 5 above)
Comment (jgoley) [inline packages/styles/README.md]: Yeah we should probably add that. Its mentioned in the resources but important enough to mention in our guide.

File: packages/styles/globals.import.styl (hunk 6 above)
Comment (jgoley) [inline packages/styles/globals.import.styl]: Right. Removed.

File: packages/styles/globals.import.styl (hunk 4 above)
Comment (jgoley) [inline packages/styles/globals.import.styl]: I've removed the margins for now and changed the font-size to ems

File: packages/styles/extends.import.styl (hunk 3 above)
Comment (jgoley) [inline packages/styles/extends.import.styl]: Good call...nib takes care of it but it doesn't add to mozilla prefix which is fine with me.

File: packages/styles/README.md (hunk 1 above)
Comment (jgoley) [inline packages/styles/README.md]: Replaced with 'Avoid styling with id attributes'

File: packages/styles/README.md [hunk 7]
```diff
@@ -1,3 +1,103 @@
...
+
+###Declaration order
+Order properties by type. Mixins, extends, functions, etc should be grouped with associated type and placed first in group.
+  1. Positioning
+  2. Display and Box-Model
+  3. Background
+  4. Font and text adjustment
+  5. Other
+  6. Media Queries
+  6. Nested Selectors
```
Comment (aslagle) [inline packages/styles/README.md]: you have two 6's

File: packages/styles/globals.import.styl [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+  height 34px
+  padding 6px 12px
+  border 1px solid #ccc
+  border-radius 4px
+  box-shadow inset 0 1px 1px rgba(0, 0, 0, .075)
+  background-color #fff
+  background-image none
+  font-size 14px
+  line-height 1.42857143
+  color #555
```
Comment (aslagle) [inline packages/styles/globals.import.styl]: is there a reason not to use variables for the textarea colors?

Comment (jgoley) [inline packages/styles/globals.import.styl]: Turns out this block isn't even necessary. I must have added it to Rana before we applied `form-control` class names to textareas. I'll remove it.

File: packages/styles/mixins.import.styl [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+  @media (-webkit-min-device-pixel-ratio: 2), (min-resolution: 192dpi), (-moz-min-device-pixel-ratio: 2)
+    background-image: url($path + '-2x.' + $type);
+
+changeBGSize($w,$h)
+  width $w
+  height $h
+  background-size $w $h
+
+centerSmall($marginBottom=0)
+  @media (max-width:400px)
```
Comment (aslagle) [inline packages/styles/mixins.import.styl]: no rupture?

Comment (jgoley) [inline packages/styles/mixins.import.styl]: Sorry, I missed that one and the retina query above.

**Comment by jgoley** [comment]
@aslagle Any idea why the import of Typographic is failing?

**Comment by aslagle** [comment]
Seems to be a problem with using it from a package. It's not available when the package is compiled, but works fine if you use it directly at the app level.

**Comment by aslagle** [comment]
Never mind - actually it's just that we have an old version of mquandalle:stylus in .meteor/versions, from before typographic was added. It works with v1.1.1.

**Comment by jgoley** [comment]
Hmm that's strange. So meteor doesn't pull the most recent version of a package when its added within packages?

**Comment by aslagle** [comment]
I'm not sure. Meteor caches local versions of packages, so maybe when you added it you already had the old one and it couldn't connect to the package server to check for an update?

When I checked out this branch, I got the older one since it was already in .meteor/versions.

**Comment by jgoley** [comment]
Maybe that was it.
I've updated the package and imported typographic. Do you think we should add anything else to the guide or base styles?

**Comment by aslagle** [comment]
Looks good to me

**Pull request merged**